src/gui/XournalView.cpp

```diff
@@ -615,14 +616,25 @@ void XournalView::zoomChanged()
 	// move this somewhere else maybe
 	layout->layoutPages();
 
-	zoom->scrollToZoomPosition(view);
+	if(zoom->isZoomPresentationMode() || zoom->isZoomFitMode())
+	{
+		scrollTo(currentPage);
+	}
+	else
+	{
+		std::tuple<double, double> pos = zoom->getScrollPositionAfterZoom(view);
```

Comment:
You can use std:tie here to be a bit more readable:
		int  scrollX , scrollY;
		std::tie( scrollX , scrollY ) = zoom->getScrollPositionAfterZoom(view);
		if( scrollX != -1 &&  scrollY != -1)
		{
			layout->scrollAbs(  scrollX , scrollY );
		}